### PR TITLE
[NixIOFr] adding support for customize annotations 

### PR DESCRIPTION
**Pull request by hkchekc**
Dealing with issue #625 

now the NixIOFr will read customize annotations

### CODE REVIEW

File: neo/rawio/nixrawio.py [hunk 1]
```diff
@@ -324,3 +355,24 @@ def _rescale_epoch_duration(self, raw_duration, dtype='float64'):
         durations = raw_duration.astype(dtype)
         # supposing unit is second, other possibilities maybe mS microS...
         return durations  # return in seconds
+
+    def _add_annotate(self, tar_ann, props, otype):
```
Comment (JuliaSprenger) [inline neo/rawio/nixrawio.py]: This function could be more general if you would use Neo object types as `otype`.

**Comment by JuliaSprenger** [comment]
Thanks for fixing this @hkchekc!
I think it would make sense to also include a couple of unittests to cover this.

**Comment by hkchekc** [comment]
@JuliaSprenger true, make sense to use the otype for some possible change later. Thanks

I will make the unitttests in another pull request.

**Comment by samuelgarcia** [comment]
Hi @JuliaSprenger and @hkchekc.

Sorry to be late on this subject.

In my mind, at rawio level there is no dependency on neo.core at all (unlike neo.io).

This is important for me.
A use case could be build an API that read ephy file directly intto pandas object (or xarrays) without neo object.
Introducing neo.core break a bit the rule.


What I anderstand here is tat ```_add_annotate``` is some kind of internal method so calling with string or real object do not matter so much.

No ?

**Comment by hkchekc** [comment]
> Hi @JuliaSprenger and @hkchekc.
> 
> Sorry to be late on this subject.
> 
> In my mind, at rawio level there is no dependency on neo.core at all (unlike neo.io).
> 
> This is important for me.
> A use case could be build an API that read ephy file directly intto pandas object (or xarrays) without neo object.
> Introducing neo.core break a bit the rule.
> 
> What I anderstand here is tat `_add_annotate` is some kind of internal method so calling with string or real object do not matter so much.
> 
> No ?

Yes, you are right, now it do not make any differences. I just thought possibly this can be used on further changes. But as you say if that contradicts with the idea of rawio, I revert the commit. Thanks for reminding

**Comment by samuelgarcia** [comment]
### Other note

For rawio developement you can use 

this internal helpers:
  * BaserawIO._raw_annotate(...)  
  * BaserawIO._repr_annotations()

And this public function:
  * BaserawIO.print_annotations()

See neo/rawio/baserawio.py.

**Comment by hkchekc** [comment]
Updates: I add some test and also fix some issues related to the events annotations in this PR. Also, I find #643 is seemingly making the cusomtized annotations for Analogsingals not feasible

**Pull request merged**